src/osd/osd_types.cc

```diff
@@ -2242,12 +2251,22 @@ void pg_pool_t::decode(ceph::buffer::list::const_iterator& bl)
     last_force_op_resend = last_force_op_resend_prenautilus;
     pg_autoscale_mode = pg_autoscale_mode_t::WARN;    // default to warn on upgrade
   }
-  if (struct_v >= 30) {
+  if (struct_v == 30) {
```

Comment:
Writing down my understanding:
old (below Squid) decoders will never even seen the changed meaning of version `>=31` as encoders use it ONLY when interacting with encoders aware about the change. The communication channel is the `features`.